player/lua/osc.lua

```diff
@@ -2975,6 +2977,7 @@ end
 opt.read_options(user_opts, "osc", function()
     validate_user_opts()
     set_osc_styles()
+    tick_delay = user_opts.tick_delay
```

Comment:
Shouldn't this check for `tick_delay_follow_display_fps` and get display_fps if needed?

Reply:
@na-na-hi 

What do you think about this? (not tested)

``` diff
diff --git a/player/lua/osc.lua b/player/lua/osc.lua
index d3085ab7a7..40d076bd15 100644
--- a/player/lua/osc.lua
+++ b/player/lua/osc.lua
@@ -93,7 +93,7 @@ local margins_opts = {
     {"b", "video-margin-ratio-bottom"},
 }
 
-local tick_delay = 1 / 60
+local tick_delay
 local tracks_osc = {}
 local tracks_mpv = {}
 local window_control_box_width = 80
@@ -2731,6 +2731,15 @@ local function update_duration_watch()
     end
 end
 
+local function set_tick_delay(_, display_fps)
+    -- may be nil if unavailable or 0 fps is reported
+    if not display_fps or not user_opts.tick_delay_follow_display_fps then
+        tick_delay = user_opts.tick_delay
+        return
+    end
+    tick_delay = 1 / display_fps
+end
+
 mp.register_event("shutdown", shutdown)
 mp.register_event("start-file", request_init)
 mp.observe_property("track-list", "native", request_init)
@@ -2779,15 +2788,8 @@ mp.observe_property("idle-active", "bool", function(_, val)
     state.idle = val
     request_tick()
 end)
-mp.observe_property("display-fps", "number", function(_, val)
-    -- may be nil if unavailable or 0 fps is reported
-    if not val or not user_opts.tick_delay_follow_display_fps then
-        tick_delay = user_opts.tick_delay
-        return
-    end
-    tick_delay = 1 / val
-end)
 
+mp.observe_property("display-fps", "number", set_tick_delay)
 mp.observe_property("pause", "bool", pause_state)
 mp.observe_property("demuxer-cache-state", "native", cache_state)
 mp.observe_property("vo-configured", "bool", request_tick)
@@ -2979,10 +2981,12 @@ local function validate_user_opts()
 end
 
 -- read options from config and command-line
-opt.read_options(user_opts, "osc", function()
+opt.read_options(user_opts, "osc", function(changed)
     validate_user_opts()
     set_osc_styles()
-    tick_delay = user_opts.tick_delay
+    if changed.tick_delay then
+        set_tick_delay("display_fps", mp.get_property_number("display_fps", nil))
+    end
     request_tick()
     visibility_mode(user_opts.visibility, true)
     update_duration_watch()
@@ -2991,6 +2995,7 @@ end)
 
 validate_user_opts()
 set_osc_styles()
+set_tick_delay()
 visibility_mode(user_opts.visibility, true)
 update_duration_watch()
 ```

Reply:
Why add `, nil` in `mp.get_property_number("display-fps", nil)`?

Reply:
> Why add , nil in mp.get_property_number("display-fps", nil)?

This is common patter used in our scripts. Feel free to change it.